src/bundle/Resources/public/js/scripts/admin.location.bookmark.js

```diff
@@ -1,42 +1,65 @@
 (function(global, doc) {
-    const SELECTOR_FORM = 'form[name="location_update_bookmark"]';
-    const SELECTOR_BOOKMARK_CHECKBOX = '#location_update_bookmark_bookmarked';
-    const SELECTOR_BOOKMARK_LOCATION_INPUT = '#location_update_bookmark_location';
+    const ENDPOINT_BOOKMARK = '/api/ezp/v2/bookmark';
     const SELECTOR_BOOKMARK_WRAPPER = '.ez-add-to-bookmarks';
     const CLASS_BOOKMARK_CHECKED = 'ez-add-to-bookmarks--checked';
+    const token = doc.querySelector('meta[name="CSRF-Token"]').content;
+    const siteaccess = doc.querySelector('meta[name="SiteAccess"]').content;
+    const bookmarkWrapper = doc.querySelector(SELECTOR_BOOKMARK_WRAPPER);
+    const currentLocationId = parseInt(bookmarkWrapper.getAttribute('data-locationid'), 10);
+    const handleRequestError = (response) => {
+        if (!response.ok) {
+            throw Error(response.statusText);
+        }
 
-    const updateBookmarkLocationInput = doc.querySelector(SELECTOR_BOOKMARK_LOCATION_INPUT);
-    const currentLocationId = parseInt(updateBookmarkLocationInput.value, 10);
-
-    const submitBookmarkForm = () => doc.querySelector(SELECTOR_FORM).submit();
-    const updateBookmarkCheckbox = (bookmarked) => {
-        const checkbox = doc.querySelector(SELECTOR_BOOKMARK_CHECKBOX);
+        return response;
+    };
+    const handleRequestResponseStatus = (response) => {
+        return handleRequestError(response).status;
+    };
+    const handleUpdateError = global.eZ.helpers.notification.showErrorNotification;
+    const updateBookmark = (addBookmark) => {
+        const method = addBookmark ? 'POST' : 'DELETE';
+        const request = new Request(`${ENDPOINT_BOOKMARK}/${currentLocationId}`, {
+            method,
+            headers: {
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
 
-        checkbox.checked = bookmarked;
+        fetch(request)
+            .then(handleRequestResponseStatus)
+            .then(toggleBookmarkIconState.bind(null, addBookmark))
+            .catch(handleUpdateError);
```

Comment:
Error messages based on `response.statusText` are not very usefult for end user. What can we do about this?